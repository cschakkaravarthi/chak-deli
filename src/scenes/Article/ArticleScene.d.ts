import React, { FC } from 'react';
import Article from '../../shared/models/Article.model';
import { match } from 'react-router';
declare type Props = {
    match?: match;
    article?: Article;
    fetchArticleById?: <T>(a: number) => Promise<T>;
    articleId?: number;
};
export declare const ArticleScene: FC<Props>;
declare const _default: import("react-redux").ConnectedComponent<React.FC<Props>, Pick<Props, never> & Props>;
export default _default;
//# sourceMappingURL=ArticleScene.d.ts.map