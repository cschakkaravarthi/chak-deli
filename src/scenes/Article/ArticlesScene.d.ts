import React, { FC } from 'react';
import Article from '../../shared/models/Article.model';
import Facet from '../../shared/models/Facet.model';
declare type Props = {
    facets?: Facet;
    isLastPage: boolean;
    setFilterFacet?: any;
    selectedFacet?: string;
    articleList?: Article[];
    clearArticlesList?: any;
    fetchArticles?: (a?: number, b?: string, c?: string) => void;
    likeArticleToggle?: (contentId: number, likedByUser: boolean) => void;
};
export declare const ArticlesScene: FC<Props>;
declare const _default: import("react-redux").ConnectedComponent<React.FC<Props>, Pick<Props, "facets" | "isLastPage" | "selectedFacet" | "articleList">>;
export default _default;
//# sourceMappingURL=ArticlesScene.d.ts.map