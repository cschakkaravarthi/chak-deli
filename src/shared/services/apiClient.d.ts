declare const client: import("axios").AxiosInstance;
export default client;
//# sourceMappingURL=apiClient.d.ts.map