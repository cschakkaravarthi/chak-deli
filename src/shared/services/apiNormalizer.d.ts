import Article from '../models/Article.model';
import Bookmark from '../models/Bookmark.model';
import BoxModel from '../models/Box.model';
import EventModel from '../models/Event.model';
import Link, { LinkGroupItems } from './../models/Link.model';
import Notification from '../models/Notification.model';
import OfficesPeople, { OfficeItem } from './../models/OfficesPeople.model';
import PeopleModel from '../models/People.model';
import Todo from '../models/Todo.model';
import { CatalogSearch } from '../models/Catalog.model';
import { ContactGroup, Department } from '../models/Department.model';
import { DocumentSearch } from '../models/Document.model';
import { KnowledgeSearch } from '../models/KnowledgeBase.model';
import { Manager, Users } from '../models/OrgChart.model';
import { Office, OfficeSearch } from '../models/Office.model';
import { Page, PageSearch } from '../models/Page.model';
import { ResponseData, SuccessfulResponse } from './apiService';
import { Categories, ManualNotification, ManualNotificationListModel, ManualNotifications, NotificationBell } from '../models/ManualNotification.model';
import UserInfoModel, { UserProfileModel, UserProfilePicture, ContactList, UserProfileUpdateResponse, ContactResponse } from '../../shared/models/UserInfo.model';
import { TaxonomyTermModel, VariousContentGroupModel, VariousContentModel } from '../types/contentTypes';
export default class ApiNormalizer {
    /**
     * Sanitation and validation of API response is crucial to prevent bugs and XSS.
     */
    static normalizeContent(response: ResponseData): VariousContentModel[];
    static normalizeGroupedContent(response: ResponseData): VariousContentGroupModel[];
    static normalizeContentSetGroup(response: ResponseData): VariousContentGroupModel | null;
    static normalizeContentSet(response: ResponseData): VariousContentModel | null;
    static validateArticle(content: ResponseData): ResponseData | null;
    static validateEvent(content: ResponseData): ResponseData | null;
    static validateDepartment(content: ResponseData): ResponseData | null;
    static validateOffice(content: ResponseData): ResponseData | null;
    static validatePeople(content: ResponseData): ResponseData | null;
    static validateNotification(content: ResponseData): ResponseData | null;
    static validateLinks(content: ResponseData): ResponseData | null;
    static validateBox(content: ResponseData): ResponseData | null;
    static validateSuccessfulResponse(content: ResponseData): ResponseData | null;
    static normalizeMinisiteLink(untrustedContent: ResponseData): Link | null;
    static normalizeSuccessfulResponse(untrustedContent: ResponseData): SuccessfulResponse | null;
    static normalizeBox(untrustedContent: ResponseData): BoxModel | null;
    /**
     * Convert API response into the data model we want to store in state.
     */
    static normalizeArticle(untrustedContent: ResponseData): Article | null;
    static normalizeTaxonomyTerm(content: ResponseData): TaxonomyTermModel | null;
    static normalizeTaxonomyTerms(content: ResponseData): TaxonomyTermModel[];
    static normalizeEvent(untrustedContent: ResponseData): EventModel | null;
    static normalizeGroupedContacts(contactsString: string): ContactGroup[];
    static normalizeDepartment(untrustedContent: ResponseData): Department | null;
    static normalizePerson(untrustedContent: ResponseData): PeopleModel | null;
    static validatePageSearch(content: ResponseData): ResponseData | null;
    static normalizePageSearch(untrustedContent: ResponseData): PageSearch | null;
    static validateOfficeSearch(content: ResponseData): ResponseData | null;
    static normalizeOfficeSearch(untrustedContent: ResponseData): OfficeSearch | null;
    static validateDocumentSearch(content: ResponseData): ResponseData | null;
    static normalizeDocumentSearch(untrustedContent: ResponseData): DocumentSearch | null;
    static validateKnowledgebaseSearch(content: ResponseData): ResponseData | null;
    static normalizeKnowledgebaseSearch(untrustedContent: ResponseData): KnowledgeSearch | null;
    static validateCatalogSearch(content: ResponseData): ResponseData | null;
    static normalizeCatalogSearch(untrustedContent: ResponseData): CatalogSearch | null;
    static normalizeOffice(untrustedContent: ResponseData): Office | null;
    static normalizeSearch(response: ResponseData): ResponseData;
    static normalizeSearchSuggestions(response: ResponseData): ResponseData;
    static normalizeNotification(untrustedContent: ResponseData): Notification | null;
    static normalizeManualNotification(untrustedContent: ResponseData): ManualNotification | null;
    static validateManualNotification(content: ResponseData): ResponseData | null;
    static dataToObjectManualNotification(contentData: ResponseData): ManualNotification;
    static normalizeManualNotificationFilterList(content: ResponseData): Categories[];
    static normalizeManualNotificationFilters(content: ResponseData): Categories[];
    static normalizeManualNotificationFilterData(untrustedContent: ResponseData): Categories | null;
    static validateManualNotificationFilterList(content: ResponseData): ResponseData | null;
    static normalizeArticles(response: ResponseData): ResponseData;
    static normalizeEvents(response: ResponseData): ResponseData;
    static normalizeFacets(responseFacets: ResponseData): ResponseData;
    static normalizeDepartments(response: ResponseData): Department[];
    static normalizeLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeQuickLink(untrustedContent: ResponseData): LinkGroupItems | null;
    static normalizeTodo(untrustedContent: ResponseData): Todo | null;
    static normalizeBookmark(untrustedContent: ResponseData): Bookmark | null;
    static normalizeMenuLinks(content: ResponseData): LinkGroupItems[];
    static normalizeUserLinks(trustedContent: ResponseData): string[];
    static normalizeTodos(trustedContent: ResponseData): Todo[];
    static normalizeBookmarks(trustedContent: ResponseData): Bookmark[];
    static normalizePeople(content: ResponseData): PeopleModel[];
    static normalizeManualNotificationList(content: ResponseData): ManualNotificationListModel;
    static normalizeManualNotifications(content: ResponseData): ManualNotifications;
    static normalizeNotificationsBell(content: ResponseData): NotificationBell | null;
    static normalizeNotificationBell(untrustedContent: ResponseData): NotificationBell | null;
    static validateNotificationBell(content: ResponseData): ResponseData | null;
    static normalizeNotificationsArchive(content: ResponseData): boolean | null;
    static normalizeNotificationArchive(untrustedContent: ResponseData): boolean | null;
    static validateNotificationArchive(content: ResponseData): ResponseData | null;
    static normalizeArticleById(content: ResponseData): Article[];
    static normalizeEventById(content: ResponseData): EventModel[];
    static validateUser(content: ResponseData): ResponseData | null;
    static userAndProfileContent(untrustedContent: ResponseData): UserInfoModel | null;
    static userProfilePictureContent(untrustedContent: ResponseData): UserProfilePicture | null;
    static userProfileContent(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUser(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeProfile(untrustedContent: ResponseData): UserInfoModel | null;
    static normalizeUsers(content: ResponseData): UserInfoModel[];
    static normalizePage(response: ResponseData): Page | null;
    static normalizeUserProfile(untrustedContent: ResponseData): UserProfileModel | null;
    static normalizeUserProfilePicture(untrustedContent: ResponseData): UserProfilePicture | null;
    static normalizeOrgChart(content: ResponseData): Users;
    static normalizeSubordinates(untrustedContent: ResponseData): Manager | null;
    static validateSubordinates(content: ResponseData): ResponseData | null;
    static normalizeOfficesPeoples(content: ResponseData): OfficesPeople[];
    static normalizeOfficesPeople(untrustedContent: ResponseData): OfficesPeople | null;
    static normalizeOfficeContents(content: ResponseData): OfficeItem[];
    static normalizeSetUserProfileData(content: any): any;
    static normalizeOfficeContent(content: ResponseData): OfficeItem | null;
    static validateOfficesPeople(content: ResponseData): ResponseData | null;
    static validateOfficeContent(content: ResponseData): ResponseData | null;
    static normalizeUserProfileUpdate(content: ResponseData): UserProfileUpdateResponse;
    static normalizeMyContacts(content: ResponseData): ContactList | null;
    static normalizeMyContactsResponse(content: ResponseData): ContactList[];
    static normalizeUsersByEmail(content: ResponseData): UserInfoModel[];
    static normalizeProfilepicDeleteResponse(content: ResponseData): ContactResponse | null;
}
//# sourceMappingURL=apiNormalizer.d.ts.map