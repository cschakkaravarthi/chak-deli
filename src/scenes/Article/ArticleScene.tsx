import React, { FC, useEffect, useState } from 'react';
import { connect } from 'react-redux';
import createDOMPurify from 'dompurify';
import { fetchArticleById } from '../../shared/actions/articleActions';
import { AppState } from '../../shared/types/genericTypes';
import get from 'lodash.get';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { formatDate } from '../../shared/services/date';
import { Card } from 'umgc_ui_library/lib';
import { match } from 'react-router';
import { TaxonomyTermModel, VariousContentModel } from '../../shared/types/contentTypes';
import Error404 from '../Errors/Error404';
import { findContentInState } from '../../utils/content';

type Props = {
  match?: match;
  content?: VariousContentModel;
  fetchArticleById?: <T>(a: number) => Promise<T>;
  articleId?: number;
};

const mapState = (state: AppState, props: Props): Props => {
  const articleId = parseInt(get(props, 'match.params.id'), 10);
  // If we clicked on this article from a previous list, then the article already exists in redux. Load it!
  // @TODO: Monitor performance of this solution. Maybe once we find them we should add them to a new set that is
  // @TODO: indexed by drupal_id for quicker lookup?
  const preLoadedArticles = findContentInState(state.contentReducers, [articleId]);
  return {
    ...props,
    articleId,
    content: preLoadedArticles[0]
  };
};

const actionCreators = {
  fetchArticleById
};

export const ArticleScene: FC<Props> = props => {
  const { content, fetchArticleById, articleId } = props;

  const [isFetching, setIsFetching] = useState<boolean>(true);

  if (!articleId) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  useEffect(() => {
    if (!content) {
      if (fetchArticleById) {
        fetchArticleById(articleId)
          .then(() => {
            setIsFetching(false);
          })
          .catch(() => {
            setIsFetching(false);
          });
      }
    }
  }, []);

  if (!content) {
    if (!isFetching) {
      return <Error404 />;
    } else {
      return null;
    }
  }

  const fullBody = get(content, 'body_full', '');

  return (
    <Container className="content-full">
      <Row>
        <Col md={12} className="p-1 mt-5">
          <Card
            type="article"
            variant="full"
            title={content.title}
            date={formatDate(get(content, 'created'))}
            imageUrl={get(content, 'image_uri.umgc_featured', '')}
            categories={get(content, 'articleCategory', [] as TaxonomyTermModel[])}
            content={
              <div
                className="position-relative"
                dangerouslySetInnerHTML={{
                  __html: createDOMPurify.sanitize(fullBody)
                }}
              />
            }
          />
        </Col>
      </Row>
    </Container>
  );
};

export default connect(mapState, actionCreators)(ArticleScene);
