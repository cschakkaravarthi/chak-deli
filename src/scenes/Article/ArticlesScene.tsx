import React, { FC, useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { filterQuery } from '../../utils/customHooks';
import { AppState } from '../../shared/types/genericTypes';
import {
  ContentQueryParamsType,
  contentToArticle,
  FacetsModel,
  skeletonBaseContentModel,
  TaxonomyTermModel,
  VariousContentModel
} from '../../shared/types/contentTypes';
import { toQueryString } from '../../utils/content';
import {
  fetchContent,
  fetchContentType,
  likeContentToggle,
  likeContentToggleType
} from '../../shared/actions/contentActions';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { Form } from 'react-bootstrap';
import { FacetProperty } from '../../shared/models/Facet.model';
import get from 'lodash.get';
import { formatDate } from '../../shared/services/date';
import { Link } from 'react-router-dom';
import { handleScrollToBottom } from '../../utils/scroller';
import { Card } from 'umgc_ui_library/lib';
import * as queryString from 'querystring';
import Skeleton from 'react-loading-skeleton';

const ARTICLE_LIMIT = 9;
const ARTICLE_FACET = 'article_category';
const ARTICLE_SORT = 'latest';

type Props = {
  facets?: FacetsModel;
  moreContentAvailable?: boolean;
  paramSet?: ContentQueryParamsType;
  content?: VariousContentModel[];
  fetchContent: fetchContentType;
  likeContentToggle?: likeContentToggleType;
};

const mapState: any = (state: AppState, props: Props): Props => {
  // First we put together a set of parameters that uniquely identify this data set. This should include any type
  // restrictions, categories, and sorts. Any time I make a query with this parameter set, I should be able to trust
  // that all the data is the same. It should NOT include paging params like limit and offset. We will stringify this
  // param set and use it as the redux key for this set of data, as well as the query string for the API call.
  // @see the ContentQueryParamsType definition for what possible keys should be. Values are comma-separated.
  const paramSet: ContentQueryParamsType = {
    type: 'article',
    facet: ARTICLE_FACET,
    sort: ARTICLE_SORT
  };
  // When a facet has been selected, it will exist in the query string for this scene. Here we load it and apply it to
  // the param set. That means this will be a whole new data set at the new stringified key.
  const chosenFacet = queryString.parse(get(props, 'location.search').substr(1))[ARTICLE_FACET];
  if (chosenFacet && typeof chosenFacet === 'string') {
    paramSet[ARTICLE_FACET] = chosenFacet;
  }
  // Here we get whatever is already loaded at the key that matches the data we require (stringified paramSet).
  const content = state.contentReducers.content[toQueryString(paramSet)];
  return {
    ...props,
    paramSet,
    content: content ? content.content : [],
    facets: content ? content.facets : undefined,
    // If the amount of content we have loaded is less than the total amount of content available for this paramSet,
    // then the pager should be active.
    moreContentAvailable: content ? content.content.length < content.totalRecords : true
  };
};

const actionCreators = {
  fetchContent,
  likeContentToggle
};

export const ArticlesScene: FC<Props> = props => {
  const { content, moreContentAvailable, facets, fetchContent, likeContentToggle, paramSet } = props;

  // This will be a tid if a facet has been selected in the query string, otherwise undefined.
  const selectedFacet = paramSet && paramSet[ARTICLE_FACET];
  const categories = get(facets, ARTICLE_FACET);

  const [isFetching, setIsFetching] = useState(true);

  const enableFetch = (): void => setIsFetching(true);
  const disableFetch = (): void => setIsFetching(false);

  // This method runs constantly as you scroll. If the pager is active, and we aren't actually fetching anything atm,
  // then check to see if the current position is "bottom", and if so, green-light the next "fetch".
  function handleScroll (): void {
    if (moreContentAvailable && !isFetching) {
      handleScrollToBottom(enableFetch);
    }
  }

  // Here we add the scroll listener. Every time isFetching changes, the listener is removed and re-added
  // with the new value.
  useEffect(() => {
    window.addEventListener('scroll', handleScroll);
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, [isFetching]);

  // Here is where we actually fetch the content. When we have a green-light to fetch (i.e. isFetching is true), then
  // pass our param set to fetchContent, along with our pager parameters to get the next page of content for this
  // paramSet. When it is complete, then disable fetching, so we don't start fetching in a loop. It will stay disabled
  // unless someone scrolls to the bottom and the listener re-enables it. At which point this will fire again.
  useEffect(() => {
    if (isFetching && fetchContent) {
      fetchContent(paramSet || {}, ARTICLE_LIMIT, content ? content.length : 0, disableFetch);
    } else {
      disableFetch();
    }
  }, [isFetching, selectedFacet]);

  // When the drop-down changes, we change the selected facet. That means we are now loading content for a new ParamSet!
  // So we will start with 0 items loaded, and load a whole new set. We assume that sorting, and available filters, and
  // number of items, could all be different for this new ParamSet.
  const setFilterQuery = filterQuery();
  function handleDropDownChange (value: string): void {
    // When the dropdown changes, redirect to the faceted data set.
    setFilterQuery.useFilterQuery(ARTICLE_FACET, value, enableFetch);
  }

  // For this component, it seems reasonable that while we are waiting on content to load we want to show a whole page
  // of skeleton cards. So we create 9 items of "empty" data to feed into the cards, and the UI Library should handle
  // skeletons from there.
  const skeletonContent = [];
  for (let i = 0; i < ARTICLE_LIMIT; i++) {
    const skeletonItem: VariousContentModel = {
      ...skeletonBaseContentModel,
      drupal_id: i,
      type: 'article'
    };
    skeletonContent.push(skeletonItem);
  }

  const loadingContent = (!content || !content.length) && isFetching;

  // Our article set will either be loaded content, or a set of skeletons.
  const articles = loadingContent ? skeletonContent : content;

  return (
    <Container className="article-list mt-5">
      <Row className="mb-2 article-list-header">
        <Col md={6}>
          <p className="font-size-xl">
            <b>News &amp; Articles</b>
          </p>
        </Col>
        <Col md={6} className="article-filter-section my-auto">
          {(categories && categories.length) || !isFetching ? (
            <Form className="d-flex justify-content-start justify-content-md-end">
              <Form.Label className="font-size-sm w-25 d-flex justify-content-start justify-content-md-end mr-sm-3 mr-0 mb-0 align-items-center">
                Filter by:
              </Form.Label>
              <Form.Control
                size="sm"
                as="select"
                id="filter"
                name="filter"
                className="w-auto rounded-0"
                onChange={(e: React.SyntheticEvent) => handleDropDownChange((e.target as HTMLInputElement).value)}
                value={selectedFacet}
              >
                <option value="">All</option>
                {categories instanceof Array &&
                  categories.map((category: FacetProperty) => (
                    <option key={category.drupal_id} value={category.drupal_id}>
                      {category.title}
                    </option>
                  ))}
              </Form.Control>
            </Form>
          ) : (
            <Skeleton width={100} />
          )}
        </Col>
      </Row>
      <Row>
        {articles &&
          articles.map((c: VariousContentModel) => {
            // In this case I expect all content to be of type article, so I convert my type, so I can use properties that
            // are specific to articles, like "likesCount". If I got back some content that didn't match the user type, it
            // would be null, so would not get rendered.
            const article = contentToArticle(c);
            return (
              article && (
                <Col md={4} key={article.drupal_id} className="mb-3 mb-md-5">
                  <Card
                    pictureSkeletonHeight={300}
                    isFetching={loadingContent}
                    type="article"
                    title={article.title}
                    imageUrl={get(article, 'image_uri.umgc_thumbnail', '')}
                    date={formatDate(article.created)}
                    categories={get(article, 'articleCategory', [] as TaxonomyTermModel[])}
                    MainLink={(p: React.PropsWithChildren<{}>) => (
                      <Link to={'article/' + article.drupal_id}>{p.children}</Link>
                    )}
                    handleThumbClick={() =>
                      likeContentToggle && likeContentToggle(article.drupal_id, !article.likedByUser)
                    }
                    likedStatus={article.likedByUser}
                    likes={article.likesCount}
                  />
                </Col>
              )
            );
          })}
      </Row>
      <Row>
        <Col>
          {isFetching && (
            <Col className="p-5 d-flex justify-content-center text-secondary">
              <strong>Fetching articles ...</strong>
            </Col>
          )}
        </Col>
      </Row>
    </Container>
  );
};

export default connect(mapState, actionCreators)(ArticlesScene);
