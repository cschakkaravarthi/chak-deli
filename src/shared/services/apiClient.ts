import axios from 'axios';
import get from 'lodash.get';
import packageJson from '../../../package.json';
import { getAccessToken } from '../../reactAuthProvider';
import { appEnv } from '../../utils/customHooks';

const apiVersion = packageJson.umgcApi.useEdge ? 'dev/' : `v${packageJson.umgcApi.version}/`;

// For some reason appEnv is undefined at this point, but window.env is not?
const client = axios.create({
  baseURL: get(appEnv || window.env, 'centralApi.baseUrl') + apiVersion
});

client.interceptors.request.use(request => {
  return new Promise((resolve, reject) => {
    getAccessToken()
      .then(accessTokenRepsonse => {
        request.headers.Authorization = 'Bearer ' + accessTokenRepsonse;
        resolve(request);
      })
      .catch(() => {
        // console.log('the catch token is ' + client.defaults.headers.common.Authorization);
        reject(request);
      });
  });
});

// @TODO: Logs for dev only.
client.interceptors.request.use(request => {
  // console.log('Starting Request:');
  // console.log(request);
  return request;
});

client.interceptors.response.use(response => {
  // console.log('Response:');
  // console.log(response);
  return response;
});

export default client;
