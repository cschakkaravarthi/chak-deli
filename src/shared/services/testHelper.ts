import { ReactElement } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import getStore from './mockGlobalStore';

export const shallowRender = (toRender: JSX.Element): ReactElement => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};

export const getMockStore = getStore;

export const drawerMockNavigation = {
  state: {
    isDrawerOpen: true,
    isTransitioning: false,
    index: 1,
    routes: [],
    key: 'abc',
    params: {}
  },
  dispatch: jest.fn(),
  goBack: jest.fn(),
  dismiss: jest.fn(),
  navigate: jest.fn(),
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  dangerouslyGetParent: jest.fn()
};
