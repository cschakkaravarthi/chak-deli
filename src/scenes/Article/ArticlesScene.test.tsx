import React from 'react';
import { ArticlesScene } from './ArticlesScene';
import { MemoryRouter } from 'react-router-dom';
import { shallowRender } from '../../shared/services/testHelper';
import { dummyArticle } from '../../shared/models/Article.model';

describe('Feed', () => {
  it.skip('renders without articles', () => {
    const tree = shallowRender(
      <MemoryRouter>
        <ArticlesScene fetchContent={jest.fn()} />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders with articles', () => {
    const tree = shallowRender(
      <MemoryRouter keyLength={0}>
        <ArticlesScene
          fetchContent={jest.fn()}
          content={[dummyArticle]}
        />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
