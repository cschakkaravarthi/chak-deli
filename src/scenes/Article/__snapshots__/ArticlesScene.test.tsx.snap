// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders with articles 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Unknown
    content={
      Array [
        Object {
          "articleCategory": Array [
            Object {
              "drupal_id": 45,
              "icon": null,
              "title": "Industry News",
              "weight": 1,
            },
          ],
          "body_full": "<p><strong>An Article!</strong>&nbsp;About&nbsp;<em>music</em>! This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content. This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.This is a paragraph with many lines of content.</p>",
          "categories": Array [
            Object {
              "drupal_id": 1,
              "icon": "test.png",
              "title": "test",
              "weight": 1,
            },
          ],
          "category_ancestors": Array [],
          "created": 1568727365,
          "drupal_id": 1,
          "image_uri": Object {
            "source": "undefined",
            "umgc_banner": "undefined",
            "umgc_embedded": "undefined",
            "umgc_featured": "undefined",
            "umgc_thumbnail": "undefined",
          },
          "likedByUser": true,
          "likesCount": 5,
          "owner": Array [
            Object {
              "drupal_id": 1,
              "icon": null,
              "title": "Corporate Communications",
              "weight": 0,
            },
          ],
          "summary": "summary content",
          "tags": Array [
            Object {
              "drupal_id": 1,
              "title": "Company news article 6",
            },
          ],
          "title": "Test Article",
          "type": "article",
        },
      ]
    }
    fetchContent={[MockFunction]}
  />
</Router>
`;
